### PR TITLE
article/cognito-demystified

### PR DESCRIPTION
**Pull request by njomb**

### CODE REVIEW

File: _posts/2021-02-24-cognito-demystified.markdown [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+## Modern Problem, Modern Solution. 🚀
+
+Say you are a cryptocurrency fanatic, and you are determined to spread your distincitive, in-depth, Google search acquired knowledge to others on the internet in the form of a subscription service (knowing the demographic of those of you reading this, this isnt far fetched). Like any good subscription service, you aim to release diluted content to the masses for free, littered with advertisements, and place the good stuff behind a paywall. Building a website on AWS delivering text based content likely will not be a challenge, although you're going to have to decide how to manage user accounts, metadata, and subscription information. Furthermore, how are you going to protect your money making content?  How are you going to do this at scale? How are you going to make the user experiencess seamless? Will Dogecoin ever reach the moon?
```
Comment (Hamo69) [inline _posts/2021-02-24-cognito-demystified.markdown:17]: > distincitive

Should be "distinctive"

Comment (Hamo69) [inline _posts/2021-02-24-cognito-demystified.markdown:17]: > experiencess

should be "experience"

File: _posts/2021-02-24-cognito-demystified.markdown [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+
+### Stop Reading, Start Doing 😤
+
+Maybe not as concise as I thought! I haven't even spoken about User Pool Groups yet...nevertheless, hopefully cognito seems a bit less scary - there's plenty of information here to get you started. The best way of learning is doing, so if Cognito sounds like it could work for you, do it. Whats the worst that could happen? 😉
+<br /><br />
+
+
+
+> *Nate takes pride in delivering outcomes for his clients using modern software development tools and processes.Through his persistent effort to learn more about how technology fits into the world around us, Nate has developed a passion for showing empathy to better understand and solve business problems. But don’t let any of this fool you, it should not be a surprise when Nate engages you in a healthy debate as to why Jeff is, and will always be the best member of The Wiggles, when Nate routinely overestimates how many donuts he can fit in his mouth, or when Nate inevitably tells you a dirty joke.*
```
Comment (Hamo69) [inline _posts/2021-02-24-cognito-demystified.markdown:93]: Perhaps add a call to action (ie. a link to our get in touch page etc )

**Review by Hamo69 — APPROVED** [review]
OK, the content is good and it reads well. It perhaps lacks a zing.. why should I care? Is there a specific problem or gotcha you faced? Could you show us the trick to solve it in code?

**Comment by njomb** [comment]
> OK, the content is good and it reads well. It perhaps lacks a zing.. why should I care? Is there a specific problem or gotcha you faced? Could you show us the trick to solve it in code?

Yeah I agree. There isn't really an opportunity for a code excerpt because it's so infrastructure focused. The specific problem I faced was unclear documentation about cognito

File: _posts/2021-02-24-cognito-demystified.markdown (hunk 2 above)
Comment (njomb) [reply _posts/2021-02-24-cognito-demystified.markdown:94]: see added call to action

**Comment by njomb** [comment]
I could make a more direct effort to call out the verbosity of the cognito docs, but not sure if that is enough of a zing either

File: _posts/2021-03-23-cognito-demystified.markdown [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
...t directory, it allows you to create, confirm, manage users accounts, and manage user authentication. Beyond basic directory features and storing user meta-data such as the names, addresses, and phone numbers of you users, User Pools provide the capability for users to sign in through third party identity providers like Google, Facebook, Amazon, Apple, through SAML identity providers, or directly into the pool itself. Additionally, you can create an AWS Lambda function that is triggered during user pool operations such as user sign-up, confirmation, and sign-in (authentication) with a [Lambda trigger](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html). You can even add authentication challenges, migrate users, and customise verification messages. To be fair, what you can achieve with User Pool configuration extends far beyond what I have described, but for the sake of keeping things simple...ish, [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpool.html) is a list of configurable parameters that use to customise your User Pool. Some of the more useful ones being:
```
Comment (stevenlaidlaw) [inline _posts/2021-03-23-cognito-demystified.markdown:33]: your*

**Pull request merged**